pyproject.toml

```diff
@@ -12,18 +12,27 @@ build-backend = "pex_build.hatchling.build"
 #   runs, which foils our metadata expansion since our README.rst contains inadvertant expansion
 #   tokens.
 # This pin low buys time to work through the issues.
-requires = ["hatchling<1.22.0"]
+requires = ["hatchling"]
 
 [tool.hatch.metadata.hooks.pex-adjust-metadata]
 expand = {"pex_version" = "version"}
 
+[tool.hatch.metadata.hooks.pex-adjust-metadata.project]
+requires-python = ">=2.7,<3.13,!=3.0.*,!=3.1.*,!=3.2.*,!=3.3.*,!=3.4.*"
+
+[tool.hatch.metadata.hooks.pex-adjust-metadata.project.urls]
+Changelog = "https://github.com/pex-tool/pex/blob/v{pex_version}/CHANGES.md"
+Documentation = "https://docs.pex-tool.org/"
+Download = "https://github.com/pex-tool/pex/releases/download/v{pex_version}/pex"
+Homepage = "https://github.com/pex-tool/pex"
+Source = "https://github.com/pex-tool/pex/tree/v{pex_version}"
+
```

Comment:
I think this effective move of the project metadata for requires-version and urls from below to here helps underline how the PEPs missed the boat here by not just saying dynamic trumps static, allowing the dynamic code to run; yet use standard static metadata as part of its inputs. This minor hoop I jumped through seems gratuitous on behalf of the spec. @ofek I know you're just doing your job implementing the spec.

Reply:
Although it is true that I'm trying to adhere to the spec, I was actually [the first](https://discuss.python.org/t/pep-621-round-3/5472/104) to implement it and was part of the design! I definitely understand your perspective and I can't go through the threads right now but there was a good reason for doing so. I think it had something to do with resolvers not being able to depend on data and thus being forever unable to use the metadata directly within source distributions and therefore require building during resolution in many cases.